source/features/next-scheduled-github-action.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+import {parseCron} from '@cheap-glitch/mi-cron';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import {getRepoURL, getRepoGQL} from '../github-helpers';
+
+const getActionsSchedules = cache.function(async (): Promise<Record<string, string> | false> => {
+	const {repository: {object: {entries: actions}}} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			object(expression: "HEAD:.github/workflows") {
+				... on Tree {
+					entries {
+						object { ... on Blob { text } }
+					}
+				}
+			}
+		}
+	`);
+	if (!actions) {
+		return false;
+	}
+
+	const schedules: Record<string, string> = {};
+	for (const action of actions) {
+		const actionYaml = action.object.text;
+		const name = /^name:\s+['"]?(.+)['"]?/m.exec(actionYaml);
+		const cron = /^\s*-\scron:\s+['"](.+)['"]/m.exec(actionYaml);
```

Comment:
As previously noted, does hit work:
```suggestion
		const name = /^name[:\s'"]+(.+)['"]?/.exec(actionYaml);
		const cron = /^\s*-\s*cron[:\s'"]+(.+)['"]/.exec(actionYaml);
```

Also

Reply:
I used the `m` flag because I wanted to anchor the regex to the beginning of the line. It works if you remove both `^` and `m`, but that means it could technically match "name: xxx" in a comment or something before the actual name declaration (same for `cron: xxx`).

Reply:
> I used the `m` flag because I wanted to anchor the regexes to the beginning of the line. It works if you remove both `^` and `m`

Oh wow I think I misunderstood the `m` flag, I thought it was the opposite. I was expecting `^` to be the beginning of the line already like it is in my editor.



> Is it valid YAML to have a space before `:`?

Yep, even a tab

Reply:
Updated suggestion: 
```suggestion
		const name = /^name[:\s'"]+(.+)['"]?/m.exec(actionYaml);
		const cron = /^[\s-]+cron[:\s'"]+(.+)['"]/m.exec(actionYaml);
```

Reply:
Perhaps the `cron` regex should be updated to include `schedule` in it, like `/schedule[:\n\s-]+cron`

Reply:
I have to admit I'm not a fan of `[:\s'"]+`. It works fine but feels a little "hackish". Something like `\s*:\s+['"]?` may be a little more verbose but it makes sense at first glance. Plus the first one matches stuff like `name::::: Action" which is a little weird. But in the end it's a matter of preference so I'll leave it up to you.

> Perhaps the cron regex should be updated to include schedule in it

I agree, imo the more specific the better in this case

Reply:
> Plus the first one matches stuff like `name::::: Action" which is a little weird

We're not here to validate YML, we just need to get the value and get out. `\s*:\s+['"]?` is way too verbose and it's wrong because it won't catch `name:Test`. Less noisy regexes are preferred. If we had to be _exact_ we'd use a YML parser, but we don't do that.

Reply:
Applied the suggestions and added `schedule[:\s-]` to the cron regex (removed the `\n` as it's already included in `\s`).